include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -0,0 +1,385 @@
+    calculatedObjectives.resize(population.size());
+    for (size_t i = 0; i < population.size(); i++)
+      calculatedObjectives[i] = arma::Col<double>(numObjectives);
+    EvaluateObjectives(population, objectives, calculatedObjectives);
+
+    // Perform fast non dominated sort on P_t ∪ G_t.
+    ranks.resize(population.size());
+    FastNonDominatedSort<MatType>(fronts, ranks, calculatedObjectives);
+
+    // Perform crowding distance assignment.
+    crowdingDistance.resize(population.size());
+
+    for (size_t fNum = 0; fNum < fronts.size(); fNum++)
+      CrowdingDistanceAssignment(fronts[fNum],
+                                 crowdingDistance,
+                                 lowerBound,
+                                 upperBound);
+
+    // Sort based on crowding distance.
+    std::sort(population.begin(), population.end(),
+              [this, ranks, crowdingDistance, population](MatType candidateP,
+                                                          MatType candidateQ)
+              {
+                size_t idxP, idxQ;
+                for (size_t i = 0; i < population.size(); i++)
+                {
+                  if (arma::approx_equal(population[i], candidateP, "absdiff", epsilon))
+                    idxP = i;
+
+                  if (arma::approx_equal(population[i], candidateQ, "absdiff", epsilon))
+                    idxQ = i;
+                }
+
+                return CrowdingOperator(idxP, idxQ, ranks, crowdingDistance);
+              }
+    );
+
+    // Yield a new population P_{t+1} of size populationSize.
+    population.resize(populationSize);
+  }
+
+  // Set the candidates from the best front as the output.
+  std::vector<MatType> bestFront;
+
+  for (size_t f: fronts[0])
+    bestFront.push_back(population[f]);
+
+  Callback::EndOptimization(*this, objectives, iterate, callbacks...);
+  return bestFront;
+}
+
+//! No objectives to evaluate.
+template<std::size_t I,
+         typename MatType,
+         typename ...ArbitraryFunctionType>
+typename std::enable_if<I == sizeof...(ArbitraryFunctionType), void>::type
+NSGA2::EvaluateObjectives(
+    std::vector<MatType>&,
+    std::tuple<ArbitraryFunctionType...>&,
+    std::vector<arma::Col<double> >&)
+{
+  // Nothing to do here.
+}
+
+//! Evaluate the objectives for the entire population.
+template<std::size_t I,
+         typename MatType,
+         typename ...ArbitraryFunctionType>
+typename std::enable_if<I < sizeof...(ArbitraryFunctionType), void>::type
+NSGA2::EvaluateObjectives(
+    std::vector<MatType>& population,
+    std::tuple<ArbitraryFunctionType...>& objectives,
+    std::vector<arma::Col<double> >& calculatedObjectives)
+{
+  for (size_t i = 0; i < populationSize; i++)
+  {
+    calculatedObjectives[i](I) = std::get<I>(objectives).Evaluate(population[i]);
+    EvaluateObjectives<I+1, MatType, ArbitraryFunctionType...>(population,
+                                                               objectives,
```

Comment:
Can we put this and next parameter in the same line in accordance with [this](https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping) style guideline